spring-kafka/src/main/java/org/springframework/kafka/listener/adapter/FilteringBatchMessageListenerAdapter.java

```diff
@@ -82,11 +82,11 @@ public void onMessage(List<ConsumerRecord<K, V>> records, @Nullable Acknowledgme
 		boolean consumerAware = this.delegateType.equals(ListenerType.ACKNOWLEDGING_CONSUMER_AWARE)
 						|| this.delegateType.equals(ListenerType.CONSUMER_AWARE);
 		/*
-		 *  An empty list goes to the listener if ackDiscarded is false and the listener can ack
-		 *  either through the acknowledgment
+		 *  An empty list goes to the listener if ackDiscarded is false, and the listener can ack
+		 *  either through the acknowledgment or through the consumer
 		 */
-		if (consumerRecords.size() > 0 || consumerAware
-				|| (!this.ackDiscarded && this.delegateType.equals(ListenerType.ACKNOWLEDGING))) {
+		if (consumerRecords.size() > 0
+				|| (!this.ackDiscarded && ((consumer != null && consumerAware) || (acknowledgment != null && this.delegateType.equals(ListenerType.ACKNOWLEDGING))))) {
```

Comment:
not an ultimate fixing (`consumerAware` is always true) but might be sufficient.